### ISSUE TITLE
[feature] blocklist support

### ISSUE THREAD

**Issue by Persei08** [butterproject/butter-desktop#584]
Hi,

I think it will be interesting to have blocklist supported in butter. (It is supported by webtorrent)

If you don't already know, In short, a blocklist is a file containing lines of IP ranges to block. The goal is to to increase privacy since there are lot of organization spying BitTorrent usage.
For example (Don't click the link if you fear to log your IP): http://iknowwhatyoudownload.com/en/peer/ keep a public history of what an ip have downloaded (Maybe someone in your neighborhood is downloading some Hacking conventions conferences :) ).

These feature is supported since long time in bittorrent clients (transmission, qbitorrent, µtorrent, ...) It was also implemented in webtorrent: https://github.com/feross/webtorrent/issues/44

*Possible way to implement it* (I presume): 
Ask the path to blocklist in settings, extract it (not sure if gzipped file are supported in webtorrent), put the filepath in webtorrent arguments

Thanks

**Comment by vankasteelj** [comment]
Could be an interresting option, but IMO should be left to the users discretion, because it's probably heavy as hell when parsing those gigantormous lists in v8

**Comment by popcornenthusiast** [comment]
BitTorrent blocklists are some useless piece of crap. No one needs to be connected to you in order to know that you are torrenting something, and so it's useless to try to block them. When you torrent something you enter in a swarm and the swarm announces everyone to everyone despite your personal preferences. BitTorrent blocklists only do harm by blocking some thousand/millions of IPs that no one knows to whom really belong, and so it blocks a lot of real peers to whom you need to be connected in order to actually transfer the files.

More info:

> PG2 is a placebo. Of course, you see some addresses blocked. The @#$%& thing blocks ONE-THIRD of the Internet address space, by its own claim.
> 
> What you do NOT know is whether the addresses this piece of @#$%& blocked were, in fact, the very peers who had the pieces you were downloading.
> 
> By inducing and then reinforcing paranoia, PG2 does more harm to torrenting than the RIAA, the MPAA and the FBI combined. I've said it before - if PG2 didn't exist, the MPAA would have invented it. Smiley
> 
> Here's just a small selection of the problems with this overhyped garbage:
> 
> [www.slyck.com]
> 
> The Media Defender internal email leak offered plenty of information for the taking. MediaDefender-Defenders said that they hoped that the email leak will prove to be a viable tool to protect against anti-P2P efforts. This is something BlueTack has been trying to do. After the email leak, a text file that compiles the complete IP (Internet Protocol) list Media Defender used while dropping fake files onto various P2P sites and networks was posted. While judging the effectiveness of these lists had proven to be an impossible task before the major leak, the effectiveness can now be tested.
> 
> Slyck began the investigation when BlueTack's 'Paranoid' IP filter blocked one of TVUnderground's new eDonkey2000 servers. A request for comment or information on the matter to BlueTack's team went unanswered. To this day, why BlueTack has blocked only one of TVUnderground's servers is unknown. In the meantime, Slyck is currently in the possession of a copy of BlueTack's IP filter lists, and further investigation into related matters appeared warranted.
> 
> According to the BlueTack website, "B.I.S.S. is a site dedicated to improving the safety and awareness of all our members and guests, providing News, Security articles, Software Reviews, Technical Support, Guides, IP Research and Free Software needed to help us keep our connections to the net and each other safe, secure, and free from unwelcome intruders."
> 
> Among the things offered are the blocklists, which have been met with either acceptance by the file-sharing community or complete rejection. Some say that the blocklists allow users to simply block any anti-filesharing company and allow users to connect with non-industry IPs. Others say that there is no way to get the right IPs before the IPs are changed to different addresses, thereby rendering the filters ineffective. It's been the subject of debate for quite some time amongst many experts with no real way to test the lists, at least until the Media Defender email leak.
> 
> The 'Paranoid' eMule IP filter was retrieved on September 27, 2007. The Level1 IP blacklist, which is supposed to block all known anti-p2p IPs, was retrieved on September 30, 2007. The idea behind getting these lists now is to offer ample time for Media Defender's now public IPs to be added to the lists for a much more effective blocklist for PeerGuardian users.
> 
> Slyck then obtained a copy of the publicly available 14.3MB compressed text file which lists all of the Media Defender's IP addresses. At this point, it became obvious that testing such a large volume of IPs would prove to be an overly time-consuming challenge, at least by hand. In order to alleviate this problem, it was best to test one particular IP range. Conveniently enough, the first range started with 116. Slyck then decided to test all of the IPs that started with the number 116.
> 
> The total number of IPs used by Media Defender starting with 116 was 1,474. Obviously, BlueTack did block all IPs that started with 116, but how many Media Defender IPs were successfully blocked? When Slyck investigated, there was a common theme that blocklists seemingly jumped over several ranges used by Media Defender. After some extensive study using the Level1 list for anti-p2p companies and the 'Paranoid' list, BlueTack would have successfully blocked 16 IPs. Thus, this sample test offered 1.09% protection against Media Defender in that range.
> 
> The IPs that were successfully blocked were: 116.255.1.109, 116.255.1.154, 116.255.1.244, 116.255.1.27, 116.255.1.52, 116.255.1.85, 116.215.157.243, 116.212.14.223, 116.199.202.170, 116.199.202.240, 116.199.207.83, 116.199.207.84, 116.199.226.78 , 116.199.227.11, 116.199.227.27, 116.199.227.67. The remaining 1,458 IPs would still be allowed through even with these two filters being used today.
> 
> While BlueTack may still perpetuate the idea that their filters are 99% effective, these latest findings will only fuel criticisms towards BlueTack's actual effectiveness. A complete test might not be possible short of creating a simple program to test every single number or spending weeks hand-testing every single Media Defender IP address. In the mean ime, it seems very apparent that BlueTack's filters have a few holes.
> 
> And that was tested against known and published addresses!! In order to catch those 16 addresses (probably by dumb luck and the law of large numbers), BlueTack also blocked more than 4 million INNOCENT addresses in the same range.
> 
> ==============================================================
> A SECOND TEST:
> 
> I am convinced that it is not authentic. I did a reverse lookup on 500 or so somewhat randomly picked addresses from the 5.3 million addresses in that list. Look at the results -- it's pretty clear that the vast majority of the items in the list are residential dynamic IP addresses from all over the world.
> 
> ===============================================================
> 
> [www.physorg.com]
> 
> Not Much Anonymity for Unprotected File-Sharers: Researchers Examine P2P Networks
> The same technology that allows easy sharing of music, movies and other content across a network also allows government and media companies easy access to who is illegally downloading that content.
> 
> "Note that it is not our intention here to examine how accurate and comprehensive these lists are, though this would be interesting and challenging future work."
> 
> 
> =============
> "after a quick look through the document, found elsewhere (PDF) (thanks again guys), all their stats are based on a couple of assumptions: that the blocklist contains no false positives, and more importantly, that it fails to contain no address that should be included. i will leave it to others to comment on the likelyhood of these assumptions being correct."
> ===============
> FROM PG;s ow website:
> 
> Well, it is accurate in the sense that it blocks everything on your blocklist.
> It is impossible to know _all_ the addresses to block
> 
> PeerGuardian is known to be incompatible with McAfee and BlackICE firewalls. Outpost is also known to cause a problem if you shut down PG2 while it is running. There is currently no way around this, so we recommend you try switching to another firewall
> 
> PeerGuardian blocked someone, should I be worried?
> Well, it was blocked, so why would you worry?
> 
> PeerGuardian is slowing down my connection!
> This occurs because of the way PeerGuardian blocks packets, not connections.
> 
> PeerGuardian is blocking an IP like crazy, should I worry?
> PeerGuardian will constantly block IPs. Many times you will see IPs get blocked three or more times before giving up - this is due to the way most computers handle reliable connecting. After a period of time, people may retry to see if you are responding yet. This does not mean people are spying on you.
> 
> A block list is a list of bad IP ranges that are know to spy on people’s computers.
> 
> 
> 
> This software is provided 'as-is', without any express or implied warranty. In no event will the authors be held liable for any damages arising from the use of this software.
> 
> Total of IP's being blocked right now:
> 
> * 2,377,645,666
> 
> 
> 
> first things first, if the government was and/or are spying on you right now, it will be on a IP range that is not on our blocklists and is a secret range of IPs, second thing is, if the government was and/or are spying on you right now, they would be doing it on a whole diffrent level.
> 
> 
> 
> ==================================
> 
> I have more. Lots more. Smiley
> 
> 
> Just felt that this will make explanations a bit easier. Smiley
> 
> In summary:
> Peerguardian does not work at all and actually damages bittorrent by blocking millions of good peers. I mean they blocked their own update servers...come on.... Roll Eyes

Source: http://forums.peerblock.com/read.php?8,11334,11336#msg-11336

**Comment by vankasteelj** [comment]
yes, that is a known fact. But if it exists and is so widely supported, it's probably that it has at least some ground... no? 
Maybe it helps for copyright companies? I don't know who else would want to check what a p2p user is doing, unless you're already under investigation. 

I have no take on this. A little protection seems better than none, even if the risk 0 doesn't exist.

Note that I'm thinking about this as the app can be used "almost" like a torrent client, and it could be a feature for some users (hence the : shouldnt be default, but a possibility)

**Comment by popcornenthusiast** [comment]
> yes, that is a known fact. But if it exists and is so widely supported, it's probably that it has at least some ground... no?

I don't think so. Most probably it's just being used by people who already don't need protection against copyright companies, hence providing a false sense of effectiveness. Pretty much only people from the US, UK and Germany (and sporadically Canada and Netherlands) are being persecuted by those companies and need some protection, so there are much more people who don't need protection but probably are also using blocklists, what weighs in favor of the effectiveness narrative.

About the fact that it's widely supported, I think that it is a collateral fact and support, and not something well thought: The BitTorrent protocol is susceptible to some attacks that can undermine all the transfers in a swarm through the use of fake peers and file transfers reset, what led to the need of implementing some IP/peer blocking measures, but then someone decided that using this feature to try to block copyright companies would be a good ideia, although it really isn't.

The IP/peer blocking thing has some historical support, though: Back in the Gnutella days the node queries for files had to be carried from peer to peer (instead of being announced to a swarm) and so that kind of blocking could be useful by trying to leave copyright companies out of the network, but this doesn't works for BitTorrent, as no one needs to be directly connected to each other in order to see the swarm. This blocking approach could be useful within trackers, as they could block peers before they see who is in the swarm, but then we would really have to know which IPs belongs to copyright companies in order to don't break the BitTorrent network. And that isn't the only problem: To use this approach for trackers we would have do dismiss the use of DHT, as it is the one which really "suffers" from the lack of privacy problem, but then we would be back to some dark days where having a tracker offline would just break the network.

> Maybe it helps for copyright companies?

They only want the users IPs in order to persecute them and they don't need to be connected to them in order to get their IPs.

> A little protection seems better than none, even if the risk 0 doesn't exist.

It just isn't a protection.

> Note that I'm thinking about this as the app can be used "almost" like a torrent client, and it could be a feature for some users (hence the : shouldnt be default, but a possibility)

I do think that Butter needs some IP/peer blocking feature, but only to block fake peers.

**Comment by Persei08** [comment]
@vankasteelj , Yes this option should be left to the user discretion to use it or not.
Also after the long reaction of @popcornenthusiast , I think this option could have a help tooltip informing about the fact the blocklist could be more or less accurate, not protecting them 100% from all spies and resulting in blocking some legitimate IP too.

**Comment by xaiki** [comment]
less UI more <3:
 - i'm all in for an automatic block-list patch
 - I'm all in for a cmd line argument block-list patch
 - I'm all in for a settings block-list patch

i'm strongly against adding UI for a blocklist

**Comment by vankasteelj** [comment]
automatic is no : performance will decrease.
cmd line is no : we're doing an app with an UI for a reason
settings : once the new settings are in, it'll be easy to add a view for it, as an advanced setting.

If we (or someone... @Persei08 i'm looking at you) implement this option, it could be something like

**Comment by Persei08** [comment]
I have no experience at all with NodeJs and OOJS but I study how things works and maybe try.

**Comment by msjaksic** [comment]
Strongly against this in any way!

We are Butter Project, we deal with only legal content. There is no need to implement this here. If our any fork wants to implement this, please do. 
This is not a place to have this.

**Comment by vankasteelj** [comment]
I have to say I don't know how the spies work with torrents. Do they track a pool of users (in which case it's useful even for privacy) or a specific torrent containing their property (in which case, well, it's indeed less "our problem")

**Comment by Persei08** [comment]
I think it's also a privacy thing even if we all know blocklists are mainly used for hiding illegal downloads.

Butter can be used like a torrent client, and IMO it should have some settings for privacy. Blocklist support is one, disabling DHT is another one I have in mind.

Now there are some questions about the spies:
Who are they? Are they legit? Which torrents are they monitoring? Are they making collected data Publicly availlable?
We don't exactly know who they are and what they are doing.

Look a bit deeper into my favorite one: iknowwhatyoudownload.com which provides list of IPs at the bottom of the pages of each monitored torrents.

It seem they mainly track illegals contents, but not only, so you can download a legal content and having your IP logged into there site.
They track:
- Linux isos ([Ubuntu](https://iknowwhatyoudownload.com/en/torrent/?infohash=33395da120c9a4758e896ded4dec5f2495c9973f), [Linux Mint](https://iknowwhatyoudownload.com/en/torrent/?infohash=76ab3f47a88b558a31a61014094eeb67e84c7ca7), [Remix OS](https://iknowwhatyoudownload.com/en/torrent/?id=8dc061830a5d3a090e2c8dd0accb3fa82d8fe0bba49453d1c1d0c325cc0d1f9b0b0b5fdfc56a57e4c1faeaf9b907d429)). I know butter isn't the best tools for linux isos :p 
- [Night of the living Dead 1968](https://iknowwhatyoudownload.com/en/torrent/?infohash=f17fb68ce756227fce325d0513157915f5634985) which is in Public Domain in USA right? look like most of the downloaders are from USA too... and they are logged :/

For me this is a good example of abuse that can happen if we let anyone spy.

**Comment by popcornenthusiast** [comment]
> informing about the fact the blocklist could be more or less accurate, not protecting them 100% from all spies

Let me say this again: Blocklists are NOT going to prevent copyright companies (or anyone else) of knowing which torrents you are downloading. It isn't that is going to prevent it sometimes, it's just that it isn't going to prevent it at all. No one needs to be connected to you in order to know what you are downloading, they just have to look at the torrents swarms and your IP will be there. If they want to persecute/prosecute you solely based on your IP, they will and they actually do it.

> and resulting in blocking some legitimate IP too.

Yeah, "some". Do you really think that copyright companies are using two **billions** IPs just to track people?

> mainly used for hiding illegal downloads.

They don't hide it.

> it should have some settings for privacy. Blocklist support is one

Again, it isn't.

> disabling DHT is another one I have in mind.

Oh, boy... The swarms, the f*cking swarms: they don't need DHT to exist, and no one needs the DHT to look at the swarms. Disabling DHT is just a great way of destroying a torrent swarm: If the trackers fails, what is going to keep a torrent running?

> Now there are some questions about the spies:
> Who are they? 

Everyone. Copyrights companies, governments, ISPs, academics, torrent search engines, people looking for fun... 

> Are they legit? 

Of course they are. Isn't a crime to look at the swarms.

> Which torrents are they monitoring? 

> Do they track a pool of users (in which case it's useful even for privacy) or a specific torrent containing their property (in which case, well, it's indeed less "our problem")

Both of it: The entire network and some millions of picked torrents, it just depends on who you are. Are you a government, ISP, academic, torrent search engine or someone looking for fun? Then you will look at all the swarms. Are you a copyright company? Then you will look at torrents of copyrighted content from your clients. But, again, all of those people only have to look at the swarms to know who is downloading/uploading what.

> Are they making collected data Publicly availlable?

You talked about iknowwhatyoudownload.com, so you already know the answer. Why are you asking for it?

Anyway, swarms are public by definition, so people may just make it more accessible and comprehensible.

> We don't exactly know who they are and what they are doing.

Yes, we do, you can even hire then.

> For me this is a good example of abuse that can happen if we let anyone spy.

It Isn't an abuse. Like I said, looking at swarms isn't a crime or some legal offense, it's just public info and everyone can see everyone. If you want privacy, pay for a VPN.

Some papers on this matter:

BitMON: A Tool for Automated Monitoring of the BitTorrent DHT: http://ieeexplore.ieee.org/document/5569978/

A Resource-Efficient Method for Crawling Swarm Information in Multiple BitTorrent Networks: http://ieeexplore.ieee.org/document/5741398/

Crawling BitTorrent DHTs for Fun and Profit: https://jhalderm.com/pub/papers/dht-woot10.pdf

Challenges and Directions for Monitoring P2P File Sharing Networks: http://dmca.cs.washington.edu/uwcse_dmca_tr.pdf

Real-World Sybil Attacks in BitTorrent Mainline DHT: https://www.cl.cam.ac.uk/~lw525/publications/security.pdf

**Comment by Persei08** [comment]
Isn't monitoring the swarm an indirect detection method? and connecting to the peer a direct detection method?
Because in [this torrentfreak news](https://torrentfreak.com/anti-piracy-blocklists-dont-keep-bittorrent-spies-out-120904/), they said:

> In the past, indirect methods – where monitoring companies obtain lists of IP-addresses **without connecting to the downloaders** – have been heavily criticized. The main problem is that these lead to a high number of false accusations. For example, research has shown that due to shoddy techniques even a network printer can be accused of sharing copyrighted files on BitTorrent.
> 
> In the paper the researchers found that **direct methods** – where the anti-piracy group **confirms that downloaders are actually sharing** – are also widely used now. Their paper is first to provide evidence of direct monitoring, suggesting that monitoring companies are upping their accuracy.

So, if I understand well. A blocklist can protect me from spies using direct detection method, no?

> This data allowed them to compare their findings to the IP-addresses that are blocked by the popular i-Blocklist blocklist, to see how effective it is at keeping BitTorrent spies out.
> 
> Perhaps not surprisingly the blocklists doesn’t offer complete security. 69% of the IP-addresses of monitoring companies were blocked, but the other 31% were not. In other words, nearly one in three logging attempts bypassed the blocklist.

And this part, Does it mean i-blocklist list blocked 69% of the IP-addresses of monitoring compagnies using direct detection method?

**Comment by popcornenthusiast** [comment]
> So, if I understand well. A blocklist can protect me from spies using direct detection method, no?

No. Let's make clear what "indirect monitoring" and "direct monitoring" (terms from the paper cited by TorrentFreak) really are and which are the consequences of both of them. Let's see:

Indirect monitoring: Your IP can be seen in the swarm by anyone (blocked by you or not) and, when a copyright company sees your IP in a swarm of some protected content, some things can happen: 1) The copyright company send a letter with a "fine" to your ISP telling that one of its users was downloading some pirate content using _this IP_ at _this time_, and your ISP has to redirect this letter to you; 2) Your ISP, after receiving the letter from the copyright company, can not only redirect this letter to you but also stop providing internet connection to you; 3) **The copyright company can make a legal complaint using solely your IP and try to start a judicial case.** It's up to a judge to accept it or not. If a judge accept it your ISP will have to tell which of its clients (you) was downloading that content at that time, then the case will continue and you will be prosecuted and called to testify. If you go to trial you may or may not be found guilty solely based on that IP. More recent cases and some judges are not willing to go ahead on a case solely based in IPs infos, **but this is up to each judge.**

Direct monitoring: Your IP can be seen in the swarm by anyone (blocked by you or not) and when a copyright company sees your IP in a swarm of some protected content it tries to connect to you to make sure that you are a real peer. If the company gets a positive ID some things can happen: 1) The copyright company send a letter with a "fine" to your ISP telling that one of its users was downloading some pirate content using _this IP_ at _this time_, and your ISP has to redirect this letter to you; 2) Your ISP, after receiving the letter from the copyright company, can not only redirect this letter to you but also stop providing internet connection to you; 3) The copyright company can make a legal complaint using solely your IP and a positive ID for pirate content downloading and try to start a judicial case. It's up to a judge to accept it or not. If a judge accept it your ISP will have to tell which of its clients (you) was downloading that content at that time, then the case will continue and you will be prosecuted and called to testify. If you go to trial you may or may not be found guilty solely based on that IP and the fact that you was downloading some protected content.

So, here is the thing: There are no laws saying how a copyright company has to behave when collecting IPs, neither there are laws saying how much info is enough info when it comes to pirate content download. It's up to each judge to say if only a IP (with or without download confirmation) is enough to prosecute someone. So, being really clear: Is direct monitoring something needed to prosecute someone? **NO.**

Now, from the conclusion of the [article](http://www.cs.bham.ac.uk/~tpc/Papers/P2PMonitor.pdf) cited by the TorrentFreak article (which was cited by you):

> Finally, we found that publicly-available blocklists, used by privacy-conscious BitTorrent users to prevent contact with monitors, contain large incidences of false positives and false negatives, and recommended that blocklists based on empirical research [18] are used over speculative ones.

> Does it mean i-blocklist list blocked 69% of the IP-addresses of monitoring compagnies using direct detection method?

Yes, it does say that 31% of the IP addresses used by copyright companies were able to make a direct connection to your IP. **;)**

This [other article](http://dmca.cs.washington.edu/uwcse_dmca_tr.pdf) (which I already cited) also says that 42% of the prefixes (like 69.xxx.xxx.xxx or 74.xxx.xxx.xxx) used by copyright companies (with each prefix accounting for "dozens" of IPs) were also able to make a direct connection to their IPs:

> To test our list of suspicious prefixes against blacklists, we obtained the latest versions of blacklists usedby the popular privacy protection software SafePeer and PeerGuardian. Of the 17 suspicious prefixes, 10 were blocked, and 8 of these, while allocated to a co-location service provider, are attributed in the blacklists to either MediaSentry or MediaDefender, copyright enforcement companies. However, seven of our suspicious prefixes (accounting for dozens of monitoring hosts) are not covered by current lists.

So, considering that there are no rules that have to be followed by the copyright companies, you are not getting any efficient protection by using blocklists.

> I even run my own openvpn server.

Well, your own vpn server is **your own** vpn server, so you aren't getting any protection from it.

> I don't ask this feature as a way for hiding my IP(s) to spies, but more as a way for hiding my activity (even legal) to spies. The difference is a bit subtle.

There is no difference as you are just wrong when you say that using blocklists is "hiding your activity" cause, again, it is not. With or without blocklists some IP is being announced in the swarm and can be seen by anyone, if its yours or from your VPN (the service which really is hiding your IP) is the real difference here.

**Comment by Persei08** [comment]
> So, here is the thing: There are no laws saying how a copyright company has to behave when collecting IPs, neither there are laws saying how much info is enough info when it comes to pirate content download. It's up to each judge to say if only a IP (with or without download confirmation) is enough to prosecute someone. So, being really clear: Is direct monitoring something needed to prosecute someone? 

I'm not ok with you. In France they have a legislation called HADOPI. The agent (spy) collecting data of copyright infringement have to be **agreed by authorities**. The **IP's have to be collected via direct monitoring** and the agent have to **proof the offender have a complete file and is sharing it**. If datas aren't collected by accredited agent or don't respect requirements, the case is dropped. 

Why they do it that way?... because the swarm could be corrupted with fake IPs (https://torrentfreak.com/popular-torrents-being-sabotaged-by-ipv6-peer-flood-150619/)

So only *direct detection method* seem accurate to tell what someone really download/share (legal or not)

> Yes, it does say that 31% of the IP addresses used by copyright companies were able to make a direct connection to your IP. ;)
Lol, you play with me. 69% isn't so useless. But this is related to iblocklist list, maybe a blocklist with better accuracy exist, maybe this paying one : http://torrent-live.org/?content

> Well, your own vpn server is your own vpn server, so you aren't getting any protection from it.
I could use it for downloading copyrighted content, Some laws only authorize to collect data about ISPs IPs not Hosting providers IPs.

**Comment by popcornenthusiast** [comment]
> I'm not ok with you. In France they have a legislation called HADOPI. 

I don't speak French, so I couldn't take a direct look at the law, so I'm gonna use [Wikipedia's article](https://en.wikipedia.org/wiki/HADOPI_law) about it on this one.

> The agent (spy) collecting data of copyright infringement have to be agreed by authorities. The IP's have to be collected via **direct monitoring** and the agent have to proof the offender have a complete file and is sharing it.

No, the IPs don't have to be collected via direct monitoring by the copyright companies. They only have to collect the IPs through indirect monitoring and then send this data to the ISPs, which **will be the ones to actually monitor their users** for copyright infringement. From the Wikipedia's article:

> **Enforcement
> 
> On receipt of a complaint from a copyright holder or representative, HADOPI may initiate a 'three-strike' procedure:
> 
> (1) **An email message is sent to the offending internet access subscriber, derived from the IP address involved in the claim.** The email specifies the time of the claim **but neither the object of the claim** nor the identity of the claimant.
> 
> **The ISP is then required to monitor the subject internet connection.** In addition, the internet access subscriber is invited to install a filter on his internet connection.**

If that's the case, how are you going to block your own ISP?

> Why they do it that way?... because the swarm could be corrupted with fake IPs

I know that can happen. Indeed, I already said it...

> Direct monitoring: Your IP can be seen in the swarm by anyone (blocked by you or not) and when a copyright company sees your IP in a swarm of some protected content it tries to connect to you to make sure that you are a real peer.

... But by the HADOPI law will be your ISP to do the monitoring, not the copyright company, and you can't block your own ISP. You can hide from them (and pretty much everyone else) using a VPN, but not block them.

> Lol, you play with me. 69% isn't so useless.

Yes, it is. Risking having to pay a fine or going to jail for 1 out of 3 movies you download is far, far away from being "protected". Not even a russian roulette game is that dumb.

But that isn't the worst problem. The bigger problem here is the lack of proper evaluation and description by those study authors when they say that "69% of the IPs are being blocked", cause that doesn't even translates to "You will get a protection of 69%". Why? Cause having "69% of the IPs blocked" is not the same as "Blocking copyright companies out of 69% of the swarms". Look at this scenario: 

You have 3 torrents, 3 different swarms and in each swarm you have 3 different copyright companies' IPs. You start downloading those 3 torrents and, looking for protection, use a blocklist. Then, **for each swarm**, you get to block 2 ("69%") out of 3 of the copyright companies' IPs. How much protection did you get? 

Exactly, none at all, cause you didn't block the copyright companies out of 2 of the 3 torrents/swarms you were downloading, you just blocked 2 out of the 3 IPs they were using in each swarm, and so you still have 1 IP for each swarm looking directly at you and this **still fits the "69% of the IPs blocked" narrative**. So, this blocking doesn't even implies that you are going to get caught "only" for 1 of the 3 movies, but for all of them. Just imagine a russian roulette game were all the tries could put a bullet in your head. Well, that's it.

> But this is related to iblocklist list, maybe a blocklist with better accuracy exist, maybe this paying one : http://peersm.com/getblocklist

Took a look at how that list works, and it uses the [Torrent-live](https://github.com/Ayms/torrent-live) approach, which is useless to determine which IPs belongs to a copyright company - although it could be useful to block sybil attacks (which is something completely unrelated but for what I already cited a paper).

It is possible to create better blocklists using the Ω-factor described on [this paper](https://www.researchgate.net/publication/220850358_Omnify_Investigating_the_Visibility_and_Effectiveness_of_Copyright_Monitors) (cited on the article cited by TorrentFreak), but, just like I have already said:

> This blocking approach could be useful within trackers, as they could block peers before they see who is in the swarm, [...]. And that isn't the only problem: To use this approach for trackers we would have to dismiss the use of DHT, as it is the one which really "suffers" from the lack of privacy problem, but then we would be back to some dark days where having a tracker offline would just break the network.

So, yeah, pretty much useless at the user-side.

> I could use it for downloading copyrighted content, Some laws only authorize to collect data about ISPs IPs not Hosting providers IPs.

WTF? Why are you using (2) VPNs, then?

**Comment by Persei08** [comment]
I think we are moving away from the initial discussion. Blocklist can be usefull for me because it can help to block some spies using direct connection method (the only one which seem accurate).

But to answer to @popcornenthusiast:

>I don't speak French, so I couldn't take a direct look at the law, so I'm gonna use Wikipedia's article about it on this one.

>>The agent (spy) collecting data of copyright infringement have to be agreed by authorities. The IP's have to be collected via direct monitoring and the agent have to proof the offender have a complete file and is sharing it.

>No, the IPs don't have to be collected via direct monitoring by the copyright companies. They only have to collect the IPs through indirect monitoring and then send this data to the ISPs, which will be the ones to actually monitor their users for copyright infringement. From the Wikipedia's article:

>**Enforcement
>
>On receipt of a complaint from a copyright holder or representative, HADOPI may initiate a 'three-strike' procedure:
>
>(1) An email message is sent to the offending internet access subscriber, derived from the IP address involved in the claim. The email specifies the time of the claim but neither the object of the claim nor the identity of the claimant.
>
>The ISP is then required to monitor the subject internet connection. In addition, the internet access subscriber is invited to install a filter on his internet connection.**

I think you're wrong. TMG (the spying compagny collecting IPs in France) use directed detection. They don't download the whole file but just a segment of it.
According to an official report from HADOPI: https://www.hadopi.fr/sites/default/files/page/pdf/rapport-znaty.pdf

page 12: (TMG is the spying compagny working with Frenchs copyright holders)
> TMG capte l'adresse IP au moment de l'établissement de la connexion (Mode connecté)

>TMG picks up the IP address **when the connection is established (Connected mode)**

page 16:
>Puis le système collecte l'adresse IP et le segment de 16 ko associé à l'oeuvre mise à disposition (time stamp et position du segment dans le fichier) (cf. Annexe 2). TMG calcule son propre hashcode du segment (SHA1). Le segment est un sous ensemble du fichier complet de l'oeuvre et constitue une preuvre démontrable.

>Then **the system collects the IP address and the 16 kb segment associated with the (made availlable work/oeuvre)** (time stamp and position of the segment in the file) (see Appendix 2). TMG computes its own segment hashcode (SHA1). **The segment is a subset of the complete file** of the oeuvre/work and **constitutes a demonstrable proof**.

For the other questions, I think they make us moving away from the initial discussion, so i don't anwser them

**Comment by Persei08** [comment]
oops I accidentaly closed the issue
